lib/fluent/plugin/out_kafka_buffered.rb

```diff
@@ -82,9 +82,18 @@ class Fluent::KafkaOutputBuffered < Fluent::BufferedOutput
 requires activesupport gem - records will be generated under fluent_kafka_stats.**
 DESC
 
-  config_param :monitoring_list, :array, :default => [],
+  config_param :monitoring, :string, :default => nil,
                :desc => "library to be used to monitor. statsd and datadog are supported"
 
+  config_param :monitoring_namespace, :string, :default => nil,
```

Comment:
I think that it would be better to use config_section like this:

```diff
diff --git a/lib/fluent/plugin/out_kafka_buffered.rb b/lib/fluent/plugin/out_kafka_buffered.rb
index ba66092..aaa21a5 100644
--- a/lib/fluent/plugin/out_kafka_buffered.rb
+++ b/lib/fluent/plugin/out_kafka_buffered.rb
@@ -78,7 +78,20 @@ requires activesupport gem - records will be generated under fluent_kafka_stats.
 DESC
 
   config_param :monitoring_list, :array, :default => [],
-               :desc => "library to be used to monitor. statsd and datadog are supported"
+               :desc => "library to be used to monitor. statsd and datadog are supported",
+               :obsoleted => "use <monitoring> section instead."
+
+  config_section :monitoring, param_name: :monitorings, multi: false do
+    config_param :target, :string, :default => nil,
+                 :desc => "library to be used to monitor. statsd and datadog are supported"
+    config_param :namespace, :string, :default => nil,
+                 :desc => "Use default of ruby-kafka"
+    config_param :host, :string, :default => nil,
+                 :desc => "Use default of ruby-kafka)"
+
+    config_param :port, :integer, :default => nil,
+                 :desc => "Use default of ruby-kafka"
+  end
 
   include Fluent::KafkaPluginUtil::SSLSettings
   include Fluent::KafkaPluginUtil::SaslSettings
@@ -192,9 +205,20 @@ DESC
       end
     end
 
-    @monitoring_list.each { |m|
-      require "kafka/#{m}"
-      log.info "#{m} monitoring started"
+    @monitorings.each{|m|
+      if m.target == 'statsd'
+        require "kafka/statsd"
+        Kafka::Statsd.namespace = m.namespace if m.namespace
+        Kafka::Statsd.host = m.host if m.host
+        Kafka::Statsd.port = m.port if m.port
+        log.info "statsd monitoring started"
+      elsif m.target == 'datadog'
+        require "kafka/datadog"
+        Kafka::Datadog.namespace = m.namespace if m.namespace
+        Kafka::Datadog.host = m.host if m.host
+        Kafka::Datadog.port = m.port if m.port
+        log.info "datadog monitoring started"
+      end
     }
   end
 ```

Reply:
That's Really Cool，Thank you for your time。